yt_dlp/extractor/tunein.py

```diff
@@ -1,244 +1,332 @@
+            'channel_follower_count': int,
             'description': 'md5:ee2c56794844610d045f8caf5ff34d0c',
-            'thumbnail': r're:https?://cdn-profiles\.tunein\.com/.+',
             'location': 'London, UK',
-            'ext': 'm4a',
             'live_status': 'is_live',
+            'thumbnail': r're:https?://.+',
         },
-        'params': {
-            'skip_download': True,
-        },
-    }]
-    _WEBPAGE_TESTS = [{
-        'url': 'https://www.martiniinthemorning.com/',
-        'info_dict': {
-            'id': 's55412',
-            'ext': 'mp3',
-            'title': 'TuneInStation video #s55412',
-        },
-        'expected_warnings': ['unable to extract hydration', 'Extractor failed to obtain "title"'],
+        'params': {'skip_download': 'Livestream'},
     }]
 
     def _real_extract(self, url):
         station_id = self._match_id(url)
-
-        webpage = self._download_webpage(url, station_id)
-        metadata = self._extract_metadata(webpage, station_id)
-
         formats, subtitles = self._extract_formats_and_subtitles(station_id)
+
         return {
             'id': station_id,
-            'title': traverse_obj(metadata, ('profiles', station_id, 'title')),
-            'description': traverse_obj(metadata, ('profiles', station_id, 'description')),
-            'thumbnail': traverse_obj(metadata, ('profiles', station_id, 'image')),
-            'timestamp': parse_iso8601(
-                traverse_obj(metadata, ('profiles', station_id, 'actions', 'play', 'publishTime'))),
-            'location': traverse_obj(
-                metadata, ('profiles', station_id, 'metadata', 'properties', 'location', 'displayName'),
-                ('profiles', station_id, 'properties', 'location', 'displayName')),
             'formats': formats,
             'subtitles': subtitles,
-            'is_live': traverse_obj(metadata, ('profiles', station_id, 'actions', 'play', 'isLive')),
+            **traverse_obj(self._call_api(station_id), ('Item', {
+                'title': ('Title', {clean_html}),
+                'alt_title': ('Subtitle', {clean_html}, filter),
+                'channel_follower_count': ('Actions', 'Follow', 'FollowerCount', {int_or_none}),
+                'description': ('Description', {clean_html}, filter),
+                'is_live': ('Actions', 'Play', 'IsLive', {bool}),
+                'location': ('Properties', 'Location', 'DisplayName', {str}, any),
+                'thumbnail': ('Image', {url_or_none}),
+            })),
         }
 
 
 class TuneInPodcastIE(TuneInBaseIE):
-    _VALID_URL = TuneInBaseIE._VALID_URL_BASE + r'/(?:podcasts/[^?#]+-|embed/player/)(?P<id>p\d+)/?(?:#|$)'
-    _EMBED_REGEX = [r'<iframe[^>]+src=["\'](?P<url>(?:https?://)?tunein\.com/embed/player/p\d+)']
-
+    IE_NAME = 'tunein:podcast:program'
+    _PAGE_SIZE = 20
+    _VALID_URL = r'https?://tunein\.com/podcasts(?:/[^/?#]+)?/[^/?#]+(?P<id>p\d+)/?(?:\?(?![^#]*(?i:\btopicid)=)[^#]*)?(?:#|$)'
```

Comment:
I'd be in favor of simplifying the regex and overriding `suitable()`
```suggestion
    _VALID_URL = r'https?://tunein\.com/podcasts(?:/[^/?#]+){1,2}(?P<id>p\d+)'
```